bpf/lib/ipv4.h

```diff
@@ -62,4 +85,66 @@ static __always_inline bool ipv4_is_fragment(struct iphdr *ip4)
 	return ip4->frag_off & bpf_htons(0xBFFF);
 }
 
+#if defined IPV4_FRAGMENTS && defined HAVE_LRU_MAP_TYPE
+static __always_inline bool ipv4_is_not_first_fragment(const struct iphdr *ip4)
+{
+	/* Ignore "More fragments" bit to catch all fragments but the first */
+	return ip4->frag_off & bpf_htons(0x9FFF);
```

Comment:
nit:
```suggestion
	return ip4->frag_off & bpf_htons(0x1FFF);
```

I'm not sure that we should assume any specific meaning from the highest order (reserved) bit here.

Reply:
This is to remain consistent with [`ipv4_is_fragment()`](https://github.com/cilium/cilium/blob/3713c0076f409fbb98cbb11664706734e4bd953f/bpf/lib/ipv4.h#L75) (“_The evil bit must be set to 0_”),which is called before `ipv4_is_not_first_fragment()`.

I can keep the assumption for both methods, or remove the existing one from `ipv4_is_fragment()` and adapt in my code if you prefer.

Reply:
Ah, it's an interesting case. What does "must be zero" mean in our context of writing packet-forwarding code?

For `ipv4_is_fragment()`, this leads to the drop-packet case. For this codepath, it's leading to the case where we... look up the IP/frag tuple and optionally pass the packet depending on whether there's a match on the tuple and the corresponding port matches policy?

I don't really know the answer. Maybe this is just yak-shaving, in which case maybe we can just disregard my comment.

Reply:
> Ah, it's an interesting case. What does "must be zero" mean in our context of writing packet-forwarding code?

I understand it as “we expect it to be zero as per the RFC”, if not zero discard the packet.

> For `ipv4_is_fragment()`, this leads to the drop-packet case. For this codepath, it's leading to the case where we... look up the IP/frag tuple and optionally pass the packet depending on whether there's a match on the tuple and the corresponding port matches policy?

Ah, right, I get it. But then with IP fragment support `ipv4_is_fragment()` will no longer be on this drop case (it is called prior to `ipv4_is_not_first_fragment()`).

> I don't really know the answer. Maybe this is just yak-shaving, in which case maybe we can just disregard my comment.

No you're probably right. I think we should take the check on the reserved bit out of `ipv4_is_fragment()` and remove it or move it to its own function (means one more `if`, one more check though :/).

Reply:
Removing it from `ipv4_is_fragment()` seems like the easiest approach here.

Reply:
Now done in preparatory commit.